src/UserFunctions.java

```diff
@@ -96,16 +107,23 @@ private void printCommands() {
     }
 
     private void createNewPuzzle(String seedWord) {
-        //TODO initialize global _puzzle variable
-        //puzzle = new Puzzle();
+        puzzle = new Puzzle(seedWord);
     }
 
     private void shuffleLetters() {
-        //TODO shuffle the letters on the display
+        //TODO shuffle the letters on the display (this doesn't actually change anything in puzzle, just the display)
     }
 
     private void showFoundWords() {
-        //TODO show the found word list using Puzzle.getFoundWords()
+        if (puzzle == null)
+            return;
```

Comment:
In the case of an early return, I'd either put the return on the same line as the if, or put it in a block (like you did above and in the suggestion).

```suggestion
        if (puzzle == null) {
            return;
        }
```